src/pvrclient-nextpvr.cpp

```diff
@@ -665,69 +678,119 @@ bool cPVRClientNextPVR::OpenRecordedStream(const kodi::addon::PVRRecording& reco
 {
-  return m_recordingBuffer->Length();
+  if (IsServerStreamingRecording())
+  {
+    return m_recordingBuffer->Length();
+  }
+  return -1;
 }
 
 bool cPVRClientNextPVR::IsTimeshifting()
 {
-  if (m_nowPlaying == Recording)
-    return false;
-  else
+  if (IsServerStreamingLive())
+  {
     return m_livePlayer->IsTimeshifting();
+  }
+  return false;
 }
 
 bool cPVRClientNextPVR::IsRealTimeStream()
 {
-  bool retval;
-  if (m_nowPlaying == Recording)
+  if (IsServerStreaming())
   {
-    retval = m_recordingBuffer->IsRealTimeStream();
-  }
-  else
-  {
-    retval = m_livePlayer->IsRealTimeStream();
+    if (m_nowPlaying == Recording)
+      return m_recordingBuffer->IsRealTimeStream();
+    else
+      return m_livePlayer->IsRealTimeStream();
   }
-  return retval;
+  return false;
 }
 
 PVR_ERROR cPVRClientNextPVR::GetStreamTimes(kodi::addon::PVRStreamTimes& stimes)
 {
-  PVR_ERROR rez;
-  if (m_nowPlaying == Recording)
-    rez = m_recordingBuffer->GetStreamTimes(stimes);
-  else
-    rez = m_livePlayer->GetStreamTimes(stimes);
-  return rez;
+  if (IsServerStreaming())
+  {
+    if (m_nowPlaying == Recording)
+      return m_recordingBuffer->GetStreamTimes(stimes);
+    else
+      return m_livePlayer->GetStreamTimes(stimes);
+  }
+  return PVR_ERROR_UNKNOWN;
 }
 
 PVR_ERROR cPVRClientNextPVR::GetStreamReadChunkSize(int& chunksize)
 {
-  PVR_ERROR rez = PVR_ERROR_NO_ERROR;
-  if (m_nowPlaying == Recording)
-    chunksize = m_settings.m_chunkRecording * 1024;
-  else if (m_nowPlaying == Radio)
-    chunksize = 4096;
-  else
-    rez = m_livePlayer->GetStreamReadChunkSize(chunksize);
-  return rez;
+  if (IsServerStreaming())
+  {
+    if (m_nowPlaying == TV)
+      return m_livePlayer->GetStreamReadChunkSize(chunksize);
+    if (m_nowPlaying == Recording)
+      chunksize = m_settings.m_chunkRecording * 1024;
+    else if (m_nowPlaying == Radio)
+      chunksize = 4096;
+    return PVR_ERROR_NO_ERROR;
+  }
+  return PVR_ERROR_UNKNOWN;
+}
+
+bool cPVRClientNextPVR::IsServerStreaming()
+{
+  if (IsServerStreamingLive(false) || IsServerStreamingRecording(false))
+  {
+    return true;
+  }
+  kodi::Log(ADDON_LOG_ERROR, "Unknown streaming state %d %d %d %d", m_nowPlaying, m_recordingBuffer->GetDuration(), m_livePlayer == nullptr);
+  return false;
+}
+
+bool cPVRClientNextPVR::IsServerStreamingLive(bool log)
+{
+  if ((m_nowPlaying == TV || m_nowPlaying == Radio) && m_livePlayer != nullptr)
+  {
+    return true;
+  }
+  if (log)
+    kodi::Log(ADDON_LOG_ERROR, "Unknown live streaming state %d %d %d", m_nowPlaying, m_recordingBuffer->GetDuration(), m_livePlayer == nullptr);
+  return false;
+}
+
+bool cPVRClientNextPVR::IsServerStreamingRecording(bool log)
+{
+  if (m_nowPlaying == Recording && m_recordingBuffer->GetDuration() > 0)
+  {
+    return true;
+  }
+  if (log)
+    kodi::Log(ADDON_LOG_ERROR, "Unknown recording streaming state %d %d %d", m_nowPlaying, m_recordingBuffer->GetDuration(), m_livePlayer == nullptr);
```

Comment:
`!m_livePlayer`

Reply:
I thought spelling out was clearer about what I wanted logged but I can change that.